code/lightna/magento-os-frontend/js/common/ShoppingCart.js

```diff
@@ -5,117 +5,193 @@ import { $, $$ } from '../lib/utils';
+        });
+        this.shoppingCartOpen.addEventListener('click', this.load.bind(this), {
+            once: true,
+        });
+        this.shoppingCartOpen.addEventListener('click', this.open.bind(this));
     }
 
-    addCartEventListeners() {
-        $$('.cjs-remove-from-cart').foreach((i, removeButton) => {
+    bindCartContentEvents() {
+        $$(this.selectors.removeFromCart).foreach((i, removeButton) => {
             const itemId = removeButton.getAttribute('data-item-id');
             removeButton.addEventListener('click', () => {
                 this.removeProduct(itemId);
             });
         });
 
-        $$('.cjs-close-minicart').foreach((i, trigger) => {
-            trigger.addEventListener('click', this.closeCart.bind(this));
+        $$(this.selectors.closeShoppingCart).foreach((i, trigger) => {
+            trigger.addEventListener('click', this.close.bind(this));
         });
     }
 
-    addProduct(component, actionTrigger) {
-        const data = UserInput.collect(component);
-        data.product = component.getAttribute('data-product-id');
-        request.post('/checkout/cart/add', data, {
-            onSuccess: (response) => {
-                this.toggleAnimation(actionTrigger, false);
-                this.updateMinicart.bind(this)();
-                // todo: refactor this; add type of message to response
-                const temp = document.createElement('div');
-                temp.innerHTML = response.messagesHtml;
-                if (temp.querySelector('.message').classList.contains('success')) {
-                    setTimeout(() => {
-                        this.openCart();
-                    }, 300);
-                } else {
-                    new PageMessage(response.messagesHtml);
-                }
+    load() {
+        if (this.loaded) {
+            return;
+        }
+
+        request
+          .post(
+            this.shoppingCartBlockUrl,
+            {},
+            {
+                onSuccess: (html) => {
+                    this.loaded = true;
+                    $('footer').insertAdjacentHTML('afterend', html);
+                    this.bindCartContentEvents();
+                },
             },
-            showMessage: false,
-        });
+          )
+          .then(() => {
+              this.open();
+          });
     }
 
-    removeProduct(itemId) {
+    update() {
+        request
+          .post(
+            this.shoppingCartBlockUrl,
+            {},
+            {
+                onSuccess: (html) => {
+                    $(this.selectors.shoppingCart).outerHTML = html;
+                    this.bindCartContentEvents();
+                    this.updateCounter();
+                },
+            },
+          )
+          .then(() => {
+              this.open();
+              this.updateMagentoCartSectionId();
+          });
+    }
+
+    open() {
+        setTimeout(() => {
+            document.body.classList.add(this.classes.cartOpen);
```

Comment:
is it good idea to introduce some global reference to body? maybe just body or body()
maybe also for classList, but I donn't see a good name

Reply:
BODY?

Reply:
maybe.. const BODY. if think about typing, uppercase isn't the most convenient.. probably body() more userfriendly...
if no good ideas we both like, then lets just skip it